listeners/txandblocknotification.go

```diff
@@ -0,0 +1,54 @@
+package listeners
+
+import (
+	"encoding/json"
+
+	"github.com/planetdecred/dcrlibwallet"
+)
+
+// TxAndBlockNotificationListener satisfies dcrlibwallet TxAndBlockNotificationListener interface contract.
+type TxAndBlockNotificationListener struct {
+	TxAndBlockNotifChan chan TxNotification
+}
+
+func NewTxAndBlockNotificationListener(txAndBlockNotif chan TxNotification) *TxAndBlockNotificationListener {
+	return &TxAndBlockNotificationListener{
+		TxAndBlockNotifChan: txAndBlockNotif,
+	}
+}
```

Comment:
This comment applies to other similar instances. I don't think the caller needs to provide a channel, it can be created here and the caller can access it using `listener.NotifChan`. In almost all cases where this kind of function has been called in this PR, the value passed in isn't special and can easily be used directly here.